### ISSUE TITLE
attack_prediction window bug when using backstab abilitie(not special)

### ISSUE THREAD

**Issue by newfrenchy83** [wesnoth/wesnoth#4475]
### Game and System Information
 - version 1.15.1
 -  built from source
 - gcc by Codeblock
 - Windows7

### Describe the bug
when i using that:

```
#define LEADERSHIP_SPECIAL_BACKSTAB
    # Canned definition of the Charge ability to be included in a
    # [specials] clause.
    [damage]
        id=backstab_leadership
        name= _ "murderous presence"
        female_name= _ "female^murderous presence"
        description= _ "All nearby allies get melee backstab."
		
        multiply=2
		cumulative=no
        active_on=offense
		affect_self=no
		affect_allies=yes
		
		[filter_student]
		[filter_weapon]
			range=melee
			[not]
				special_active=backstab
			[/not]
		[/filter_weapon]
		[/filter_student]
		[filter_opponent]
            formula="
                enemy_of(self, flanker) and not flanker.petrified
            where
                flanker = unit_at(direction_from(loc, other.facing))
            "
        [/filter_opponent]
		[affect_adjacent]
		[/affect_adjacent]
    [/damage]
#enddef
```

thue attack_prediction show value not multiplied then what damage value multipied by 2 in fight or then value multiplied by 2 then what unit fight with simple value of dame, or then show value of damage what it wait. using [filter_student] and [filter_opponent] in same time must cause that.

### To Reproduce

1. Start a scenario with a unit own this ability in debug mode
2. test a unit adjacent(create in debug mode) in condition of backstab respected(figher, opponent, ally in opposite) or backstab no respected
3. Compare damage value in attack_prediction window and in real fight.
4. if damage value is correct, repeat steps2 and 3 2 or 3 time.

### Expected behavior
Normally value in window and real figt must be the same.

**Comment by jostephd** [comment]
What happens here is that the Attack Enemy dialog considers the ability inactive because the units' facings haven't been set yet. The `facing` attribute is [unsynced](https://wiki.wesnoth.org/OOS_(Out_of_Sync)#List_of_Non_Mp.2FReplay_safe_Wml.2FLua_functions), and anyway there isn't exactly a strong case for expecting unit facings to be changed just because the Attack Enemy dialog was opened, so I think this is a bug in your ability, you shouldn't code the ability in terms of other.facing.

**Comment by jostephd** [comment]
You can ask on the forums for alternative ways to implement that ability.

**Comment by newfrenchy83** [comment]
Backstab option is deprecated, else i could use that insteas.

**Comment by newfrenchy83** [comment]
@jostephd no problem with backstab option, it will be removed in 1.16?

**Comment by jostephd** [comment]
I don't know. Look it up on the wiki, or use it with `--log-info=deprecation` and see if you get deprecation warnings.

**Comment by newfrenchy83** [comment]
it is write deprecated in the wiki https://wiki.wesnoth.org/AbilitiesWML#Extra_tags_and_keys_used_by_weapon_special_abilities, i will must resolve my problem by using of latent weapon like DUGI in LOTI( then what this is that who give to me the idea of special abilities in the base)

**Comment by jostephd** [comment]
I hadn't realized that the wiki for backstab [specifically recommended](https://wiki.wesnoth.org/AbilitiesWML#Common_keys_and_tags_for_specials_with_a_value) to use [that particular formula](https://github.com/wesnoth/wesnoth/blob/bdf5421fd5ba72aaa7eebaa38213da7cd7d2d878/data/core/macros/abilities.cfg#L302-L308).

I'm not sure what exactly is going on here... @CelticMinstrel, do you see why the formula from the WEAPON_SPECIAL_BACKSTAB definition doesn't work in this case?

**Comment by CelticMinstrel** [comment]
What exactly happened with this? Was it fixed (maybe in #4678?), or is there another more recent PR or issue about it? I think the correct answer would be to either update the facing when opening the attack dialog _or_ at least calculate values for the attack dialog _as if_ the facing had been correctly updated.

**Comment by newfrenchy83** [comment]
@celticminstrel it is already fixed in https://github.com/wesnoth/wesnoth/commit/ce4fb821737157761df0bab4d13e770a0efb23ec
